### PR TITLE
RFR: Print exit_code when operations fail on CLI commands

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
```
(virtualenv)~/stanley git:STORM-960/exit_code_when_fail ❯❯❯ st2 rule create -j bad_rule.json                 ✭ ✱ ◼
ERROR: Expecting property name: line 2 column 3 (char 4)
(virtualenv)~/stanley git:STORM-960/exit_code_when_fail ❯❯❯ echo $?                                        ⏎ ✭ ✱ ◼
2
(virtualenv)~/stanley git:STORM-960/exit_code_when_fail ❯❯❯ st2 action get twilio.get_sms                    ✭ ✱ ◼
Action "twilio.get_sms" is not found.

(virtualenv)~/stanley git:STORM-960/exit_code_when_fail ❯❯❯ echo $?                                        ⏎ ✭ ✱ ◼
2
(virtualenv)~/stanley git:STORM-960/exit_code_when_fail ❯❯❯

```

### CODE REVIEW

**Comment by lakshmi-kannan** [comment]
Since we have run_and_print for commands, I went with raising a special 'OperationFailureException' so we don't print the ERROR again in shell.py and return a different exit code than a generic exception. Ideally, it would be nice to map an exit code to each exception. This would mean we have to create those exception classes. I was not sure if that's the right approach. Anyways, feedback appreciated.

**Comment by Kami** [comment]
LGTM.

On an actual exception inside the client or cli code we will still print the correct traceback if debug mode is used, right? (just thinking out loud)

**Comment by Kami** [comment]
Also `docs/source/cli.rst`. It's not don't if it's not documented :package: 

Just add a new section somewhere around the top about the client exit codes.

**Comment by lakshmi-kannan** [comment]
>On an actual exception inside the client or cli code we will still print the correct traceback if debug mode is used, right? (just thinking out loud)

Yes, it works. I checked.
https://gist.github.com/lakshmi-kannan/004404121aba865f5bd1

File: st2client/st2client/commands/resource.py [hunk 1]
```diff
@@ -314,9 +316,15 @@ def run(self, args, **kwargs):
...
-                          attributes=['all'], json=args.json)
+        try:
+            instance = self.run(args, **kwargs)
+            if not instance:
+                raise Exception('Server did not create instance.')
+            self.print_output(instance, table.PropertyValueTable,
+                              attributes=['all'], json=args.json)
+        except Exception as e:
+            print('ERROR: %s' % e.message)
+            raise OperationFailureException('Create failed.')
```
Comment (Kami) [inline st2client/st2client/commands/resource.py]: Maybe we should also include original error message in the derived exception message.

File: st2client/st2client/commands/resource.py [hunk 2]
```diff
@@ -358,8 +366,12 @@ def run(self, args, **kwargs):
...
     def run_and_print(self, args, **kwargs):
         instance = self.run(args, **kwargs)
-        self.print_output(instance, table.PropertyValueTable,
-                          attributes=['all'], json=args.json)
+        try:
+            self.print_output(instance, table.PropertyValueTable,
+                              attributes=['all'], json=args.json)
+        except Exception as e:
+            print('ERROR: %s' % e.message)
+            raise OperationFailureException('Update failed.')
```
Comment (Kami) [inline st2client/st2client/commands/resource.py]: Same here.

**Comment by Kami** [comment]
Besides the two comments, LGTM, +1

**Pull request merged**